Mathlib/AlgebraicGeometry/Restrict.lean

```diff
@@ -347,7 +347,6 @@ theorem morphismRestrict_app {X Y : Scheme.{u}} (f : X ⟶ Y) (U : Opens Y) (V :
     Hom.app_eq_appLE, comp_appLE, ofRestrict_appLE, Hom.appLE_map, eqToHom_op,
     restrict_presheaf_map, eqToHom_unop] at this
   have e : ιOpens U ⁻¹ᵁ (ιOpens U ''ᵁ V) = V := by
-    congr 1
     exact Opens.ext (Set.preimage_image_eq _ Subtype.coe_injective)
```

Comment:
Can you also remove the `by exact`?

Reply:
It looks like it, thanks!

Maybe, there should be an "unnecessary `by exact ...` linter".

Reply:
It's a tough one to write since `by exact` is not quite a no-op (it delays elaboration)

Reply:
To first approximation, it could work like the "unused `set_option ... in`" linter: `try` to elaborate the proof with `by exact x` replaced by `x` and if no error is thrown, then suggest the alternative.

With more effort, there could also be a "speed" test, but I have not played around with that enough.